developers/weaviate/client-libraries/typescript/v2_v3_migration.md

```diff
@@ -129,132 +101,308 @@ Once it has been instantiated, you will notice that the client API is different
-import SearchSimilarityCodeV3 from '!!raw-loader!/_includes/code/howto/search.similarity-v3.py';
+result = await myCollection.query.nearText(['animals in movies'],{
+  limit: 2,
+  returnProperties: ['question', 'answer'],
+  returnMetadata: ['distance']
+})
 
-<Tabs groupId="languages">
-  <TabItem value="py" label="Python (v4)">
-    <FilteredTextBlock
-      text={SearchSimilarityCode}
-      startMarker="# GetNearTextPython"
-      endMarker="# END GetNearTextPython"
-      language="python"
-    />
-  </TabItem>
-
-  <TabItem value="py3" label="Python (v3)">
-    <FilteredTextBlock
-      text={SearchSimilarityCodeV3}
-      startMarker="# GetNearTextPython"
-      endMarker="# END GetNearTextPython"
-      language="python"
-    />
-  </TabItem>
+console.log(JSON.stringify(result.objects, null, 2));
+```
+
+</TabItem>
+<TabItem value="jsv2" label="JS/TS (v2)">
+
+```ts
+let result;
+
+result = await client.graphql
+  .get()
+  .withClassName('JeopardyQuestion')
+  .withNearText({ concepts: ['animals in movies'] })
+  .withLimit(2)
+  .withFields('question answer _additional { distance }')
+  .do();
+
+
+console.log(JSON.stringify(result, null, 2));
+```
+
+</TabItem>
 </Tabs>
 
 This makes it easier to understand and use. Additionally, some parameters typed (e.g. `MetadataQuery`) which makes it easier to use and reduces errors.
 
-### Seperate packages for Web and Node platforms
+### Node only Support
+
+To support gRPC, we've chosen to break up the new TypeScript client into Node and Web versions. The addition of gRPC to Weaviate means we now have the HTTP/2 protocol to contend with and we quickly discovered that gRPC and HTTP/2 don't play nicely with browser. That said, the `v3` client leverages Node.js-specific capabilities for building server applications, APIs, and CLIs, with the full gRPC functionality available.
 
-  - talk a bit more about this and package specifics 
-  - Introduction of gRPC?
+For Web use, we recommend using the `v2` client or use the `v3` [web client](#).
 
 ### Improved Functionality
 
-(with examples)
+#### Improved Bulk Insertion
+We've added an `insertMany()` method as a replacement for `objectBatcher()` to help make batch insertions easier. 
+
+<Tabs groupId="languages">
+<TabItem value="jsv3" label="JS/TS (v3)">
+
+```ts
+const questions = client.collections.get("CollectionName")
+const dataObject = [...]; // your data
+
+await questions.data.insertMany(dataBatch);
+```
+
+</TabItem>
+<TabItem value="jsv2" label="JS/TS (v2)">
+
+```ts
+let className = 'CollectionName';  // Replace with your collection name
+let dataObjs = [...];
+
+let batcher5 = client.batch.objectsBatcher();
+for (const dataObj of dataObjs)
+  batcher5 = batcher5.withObject({
+    class: className,
+    properties: dataObj,
+  });
+
+// Flush
+await batcher5.do();
+```
+
+</TabItem>
+</Tabs>
+
+For more information on batching, I recommend looking at the [batching section](typescript-v3.md#) of our client page.
```

Comment:
Consider:

For more information on batching, see [Batching](typescript-v3.md#).


That link looks broken, does it resolve properly?